skorch/net.py

```diff
@@ -1266,6 +1266,15 @@ def _get_params_for_optimizer(self, prefix, named_parameters):
     def _get_param_names(self):
         return self.__dict__.keys()
 
+    def _get_param_names_new(self):
+        # TODO: This will be the new behavior for _get_param_names in
+        # a future release. This is to make get_params work as in
+        # sklearn, i.e. not returning "learned" attributes (ending on
+        # '_'). Once the transition period has passed, remove the old
+        # code and use the new one instead.
+        return (k for k in self.__dict__
+                if not k.endswith('_') and k != 'history')
```

Comment:
`history` is... weird parameter.

Reply:
Maybe we should make `history` a property and store the values in `self.history_`?

Reply:
I made this change.